### PR TITLE
Users

### PR DESCRIPTION
**Pull request by samfolo**
Password encryption, testing precedents

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,20 +1,24 @@
 class PostsController < ApplicationController
   def new
+    # p "New: #{params}"
```
Comment (ffgi-es) [inline app/controllers/posts_controller.rb]: If all we've done is add commented out debgug prints, I don't think we should include the file in the commit as we'll just have to delete those later anyway.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  def show
+    @current_user = User.find(params[:id])
+  end
+
+  private 
+
+  def user_params
+    params.permit(:first_name, :last_name, :email, :password)
+    # p "------> #{params}"
```
Comment (ffgi-es) [inline app/controllers/users_controller.rb]: If this is working, can we get rid of the commented out debug messages?

File: config/routes.rb [hunk 3]
```diff
@@ -3,4 +3,8 @@
   root 'posts#index'
 
   resources :posts
+  resources :users
+
+  post 'users/new' => 'users#create'
```
Comment (ffgi-es) [inline config/routes.rb]: To follow RESTful convention, I think this should be `post 'users' => 'users#create'` without the new. I think that will also have already been defined when you put `resources :users`

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+  describe 'POST #create' do
+    it 'renders the page without error and redirects to their profile' do
+      post :create, params: { first_name: 'Jonathan', last_name: 'Palma', email: 'jonathan@gmail.com', password: 1234 }
+      expect(User.find_by({ email: 'jonathan@gmail.com' })).to be
+      # post :create, first_name: 'Jonathan', last_name: 'Palma', email: 'jonathan@gmail.com', password: 1234
```
Comment (ffgi-es) [inline spec/controllers/users_controller_spec.rb]: Again, we should remove comments of work in progress if we've solved the issue and it's ready to be commited.

Comment (samfolo) [reply spec/controllers/users_controller_spec.rb]: will do

**Pull request merged**